haystack/agents/conversational.py

```diff
@@ -1,63 +1,121 @@
-from typing import Optional, Callable, Union
+from typing import Optional, Dict, Any, List
+import logging
 
-from haystack.agents import Agent
+from haystack.errors import AgentError
+from haystack.agents.base import Tool, ToolsManager
+from haystack.agents import Agent, AgentStep
 from haystack.agents.memory import Memory, ConversationMemory
-from haystack.nodes import PromptNode, PromptTemplate
+from haystack.nodes import PromptNode
+
+logger = logging.getLogger(__name__)
+
+
+def agent_without_tools_parameter_resolver(query: str, agent: Agent, **kwargs) -> Dict[str, Any]:
+    """
+    A parameter resolver for ReAct based agents without tools that returns the query, the history.
+    """
+    return {"query": query, "history": agent.memory.load()}
+
+
+def conversational_agent_parameter_resolver(
+    query: str, agent: Agent, agent_step: AgentStep, **kwargs
+) -> Dict[str, Any]:
+    """
+    A parameter resolver for ReAct based agents that returns the query, the tool names, the tool names
+    with descriptions, the history of the conversation, and the transcript (internal monologue).
+    """
+    return {
+        "query": query,
+        "tool_names": agent.tm.get_tool_names(),
+        "tool_names_with_descriptions": agent.tm.get_tool_names_with_descriptions(),
+        "transcript": agent_step.transcript,
+        "history": agent.memory.load(),
+    }
 
 
 class ConversationalAgent(Agent):
     """
-    A conversational agent that can be used to build a conversational chat applications.
+    A ConversationalAgent is an extension of the Agent class that supports the use of tools in
+    conversational chat applications. This agent can make use of the ToolsManager to manage a set of tools and seamlessly integrate them into the conversation.
 
-    Here is an example of how you can create a simple chat application:
+    If no tools are provided, the Agent will be initialized with necessary arguments to have a basic chat application.
+
+    Here is an example of how you can create a chat application with tools:
     ```
     import os
 
-    from haystack.agents.base import ConversationalAgent
-    from haystack.agents.memory import ConversationSummaryMemory
+    from haystack.agents.conversational import ConversationalAgent
     from haystack.nodes import PromptNode
+    from haystack.tools import ToolsManager, Tool
+
+    # Initialize a PromptNode and a ToolsManager with the desired tools
+    prompt_node = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
+    tools_manager = ToolsManager([Tool(name="ExampleTool", pipeline_or_node=example_tool_node)])
 
-    pn = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
-    agent = ConversationalAgent(pn, memory=ConversationSummaryMemory(pn))
+    # Create the ConversationalAgent instance
+    agent = ConversationalAgent(prompt_node, tools_manager=tools_manager)
```

Comment:
This needs to be updated, `ConversationalAgent` takes a list of tools, not a tools_manager.